xapian-core/include/xapian/weight.h

```diff
@@ -590,6 +590,101 @@ class XAPIAN_VISIBILITY_DEFAULT BM25Weight : public Weight {
     double get_maxextra() const;
 };
 
+/// Xapian::Weight subclass implementing the BM25+ probabilistic formula.
+class XAPIAN_VISIBILITY_DEFAULT BM25PlusWeight : public BM25Weight {
+
+  public:
+    /** Construct a BM25PlusWeight.
```

Comment:
Conventionally we don't have a blank line between the start of the class and `public:` like you have here in the Xapian codebase.